source/DTAFunctions.c

```diff
@@ -6,14 +6,28 @@
 #include <stddef.h>
 #include <stdio.h>
 #include <string.h>
+#include <ctype.h>
+#include <stdlib.h>
 #include "ports.h"
 #include "GlobalSymbols.h"
 #include "config.h"
 #include "net_events.h"
 #include "rb3/Data.h"
 #include "rb3/SongMetadata.h"
 #include "rb3/BandSongMgr.h"
+#include "MusicLibrary.h"
 #include "rb3enhanced.h"
+#include "rb3/RockCentralGateway.h"
+
+typedef struct _song_list_vector {
+    int *start;
+    int *end;
+    int unk1;
+} song_list_vector;
+
+static song_list_vector dta_song_list = { 0 };
```

Comment:
this should probably go on the stack but im not familiar enough with this codebase to give any specific guidance here

deferring to @InvoxiPlayGames

Reply:
turns out this prob isnt a uaf since its just a vector of song ids, but it should prob be fixed anyway as it'll have song ids that no longer exist after deleting a song